configure.ac

```diff
@@ -1973,6 +1973,20 @@ XFreeEventData(display, cookie);
             if test x$definitely_enable_video_x11_xinput = xyes; then
                 SUMMARY_video_x11="${SUMMARY_video_x11} xinput2"
                 AC_DEFINE(SDL_VIDEO_DRIVER_X11_XINPUT2, 1, [ ])
+                AC_MSG_CHECKING(for xinput2 scrollinfo)
+                have_xinput2_scrollinfo=no
+                AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+                    #include <X11/Xlib.h>
+                    #include <X11/Xproto.h>
+                    #include <X11/extensions/XInput2.h>
+                    ]], [[
+XIScrollClassInfo *s;
+                    ]])], [
+                    have_xinput2_scrollinfo=yes
+                    AC_DEFINE([SDL_VIDEO_DRIVER_X11_XINPUT2_SUPPORTS_SCROLLINFO], 0, [ ])
```

Comment:
- The defined value should be `1`, not 0

- The following should be applied to SDL_config.h.in:
```diff
diff --git a/include/SDL_config.h.in b/include/SDL_config.h.in
index 195b988..f312ed0 100644
--- a/include/SDL_config.h.in
+++ b/include/SDL_config.h.in
@@ -397,6 +397,7 @@
 #undef SDL_VIDEO_DRIVER_X11_XINERAMA
 #undef SDL_VIDEO_DRIVER_X11_XINPUT2
 #undef SDL_VIDEO_DRIVER_X11_XINPUT2_SUPPORTS_MULTITOUCH
+#undef SDL_VIDEO_DRIVER_X11_XINPUT2_SUPPORTS_SCROLLINFO
 #undef SDL_VIDEO_DRIVER_X11_XFIXES
 #undef SDL_VIDEO_DRIVER_X11_XRANDR
 #undef SDL_VIDEO_DRIVER_X11_XSCRNSAVER
```